mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/ComputeQueryImpl.java

```diff
@@ -53,30 +54,49 @@ public Object execute() {
         String keyspace = theGraph.getKeyspace();
 
         Analytics analytics = typeIds.map(ids -> {
-            Set<Type> types = ids.stream().map(theGraph::getType).collect(toSet());
+            Set<Type> types = ids.stream().map(id -> {
+                Type type = theGraph.getType(id);
+                if (type == null) throw new IllegalArgumentException(ErrorMessage.ID_NOT_FOUND.getMessage(id));
+                return type;
+            }).collect(toSet());
             return new Analytics(keyspace, types);
         }).orElseGet(() ->
             new Analytics(keyspace)
         );
 
-        switch (computeMethod) {
-            case "count": {
-                return analytics.count();
-            }
-            case "degrees": {
-                return analytics.degrees();
-            }
-            case "degreesAndPersist": {
-                try {
+        try {
+            switch (computeMethod) {
+                case "count": {
+                    return analytics.count();
+                }
+                case "degrees": {
+                    return analytics.degrees();
+                }
+                case "degreesAndPersist": {
                     analytics.degreesAndPersist();
-                } catch (ExecutionException | InterruptedException e) {
-                    throw new RuntimeException(e);
+                    return "Degrees have been persisted.";
+                }
+                case "max": {
+                    return analytics.max();
+                }
+                case "mean": {
+                    return analytics.mean();
+                }
+                case "min": {
+                    return analytics.min();
+                }
+                case "std": {
+                    return analytics.std();
+                }
+                case "sum": {
+                    return analytics.sum();
+                }
+                default: {
```

Comment:
Is the default really needed ? Is it possible for `computeMethod` to be invalid ? If `computeMethod` is always internally set then I don't think the default is needed.